inference_perf/loadgen/load_generator.py

```diff
@@ -94,6 +94,7 @@ async def loop(self) -> None:
         event_loop = get_event_loop()
         item = None
         timeout = 0.5
+        session = self.client.new_session()
```

Comment:
We should be able to just have the client manage the session and have the loadgen continue utilizing the client.

The client should have a session object that is initialized once and reused via process_request.

So with your current implementation, Client process_request would look like

```
async def process_request(self, data: InferenceAPIData, stage_id: int, scheduled_time: float) -> None:
        await self._session.process_request(data, stage_id, scheduled_time)
```

Reply:
I could do this change, but my main concern would be that we're basically relying on the assumption that the multiprocess worker's variables are copied over, so a mutation of the client's state (changing `self._session`) doesn't get carried over to the parent thread, so it's a bit scary.

Reply:
> assumption that the multiprocess worker's variables are copied over

that is the fork strategy used by asyncio + multiprocess afaik

> so a mutation of the client's state (changing self._session) doesn't get carried over to the parent thread

I am not 100% sure I understand what the concern is here